src/main/java/org/slf4j/impl/Slf4jLogger.java

```diff
@@ -297,36 +491,109 @@ public void error(final String format, final Object... arguments) {
             return;
         }
         final FormattingTuple formattingTuple = MessageFormatter.arrayFormat(format, arguments);
-        log(org.jboss.logmanager.Level.ERROR, formattingTuple.getMessage(), formattingTuple.getThrowable(), arguments);
+        log(null, org.jboss.logmanager.Level.ERROR, formattingTuple.getMessage(), formattingTuple.getThrowable(), arguments);
     }
 
     @Override
     public void error(final String msg, final Throwable t) {
         if (ALT_ERROR_INT < logger.getEffectiveLevel()) {
             return;
         }
-        log(org.jboss.logmanager.Level.ERROR, msg, t);
+        log(null, org.jboss.logmanager.Level.ERROR, msg, t);
+    }
+
+    @Override
+    public boolean isErrorEnabled(Marker marker) {
+        return isErrorEnabled();
+    }
+
+    @Override
+    public void error(Marker marker, String msg) {
+        if (ALT_ERROR_INT < logger.getEffectiveLevel()) {
+            return;
+        }
+        log(marker, org.jboss.logmanager.Level.ERROR, msg, null);
+    }
+
+    @Override
+    public void error(Marker marker, String format, Object arg) {
+        if (ALT_ERROR_INT < logger.getEffectiveLevel()) {
+            return;
+        }
+        final FormattingTuple formattingTuple = MessageFormatter.format(format, arg);
+        log(marker, org.jboss.logmanager.Level.ERROR, formattingTuple.getMessage(), formattingTuple.getThrowable(), arg);
     }
 
     @Override
+    public void error(Marker marker, String format, Object arg1, Object arg2) {
+        if (ALT_ERROR_INT < logger.getEffectiveLevel()) {
+            return;
+        }
+        final FormattingTuple formattingTuple = MessageFormatter.format(format, arg1, arg2);
+        log(marker, org.jboss.logmanager.Level.ERROR, formattingTuple.getMessage(), formattingTuple.getThrowable(), arg1, arg2);
+    }
+
+    @Override
+    public void error(Marker marker, String format, Object... arguments) {
+        if (ALT_ERROR_INT < logger.getEffectiveLevel()) {
+            return;
+        }
+        final FormattingTuple formattingTuple = MessageFormatter.arrayFormat(format, arguments);
+        log(marker, org.jboss.logmanager.Level.ERROR, formattingTuple.getMessage(), formattingTuple.getThrowable(), arguments);
+    }
+
+    @Override
+    public void error(Marker marker, String msg, Throwable t) {
+        if (ALT_ERROR_INT < logger.getEffectiveLevel()) {
+            return;
+        }
+        log(null, org.jboss.logmanager.Level.ERROR, msg, t);
+    }
+
     protected Slf4jLogger readResolve() throws ObjectStreamException {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Se: The readResolve method must be declared with a return type of Object. (SE_READ_RESOLVE_MUST_RETURN_OBJECT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#se-the-readresolve-method-must-be-declared-with-a-return-type-of-object-se-read-resolve-must-return-object).
In order for the readResolve method to be recognized by the serialization mechanism, it must be declared to have a return type of Object.

Reply:
good catch! thank you!